src/main/java/com/kodilla/ecommercee/mapper/CartMapper.java

```diff
@@ -29,10 +33,14 @@ public CartDto mapToCartDto(Cart cart) {
     }
 
     public ItemDto mapToItemDto(Item item){
-        return new ItemDto(mapToCartDto(item.getCart()),productMapper.mapToProductDto(item.getProduct()));
+        return new ItemDto(item.getId(),item.getId(),item.getProduct().getId(),item.getQuantity(),item.getPrice());
+
     }
     public Item mapToItem(ItemDto itemDto){
-        return new Item(mapToCart(itemDto.getCartId()),productMapper.mapToProduct(itemDto.getProductId()),
+        return new Item(itemDto.getId(),(cartRepository.findById(itemDto.getCarId()).orElseThrow(()->
+                new EntityNotFoundException("Item with id "+ itemDto.getCarId() + " was not found."))),
+                (productRepository.findById(itemDto.getProductId()).orElseThrow(()->
+                        new EntityNotFoundException("Product with id "+ itemDto.getProductId() + " was not found."))),
                 itemDto.getQuantity(),itemDto.getPrice());
```

Comment:
- Działa, ale taki zapis nie może być, jest nieczytelne. Rozbij to na:
Cart cart = cartRepository.findById......
Product product = productRepository.findById....

return new Item(cart, product, ......)
- Nie dajesz spacji po przecinkach (w zasadzie wszędzie)
- Komunikaty błędów jako stałe wyrzuć do klasy np:
private static final String NO_CART = "Cart not found, id: ";
potem EntityNotFoundException(NO_CART + id)

Reply:
Będą enumy w przyszłości we wszystkim to się pozmienia hurtowo (to chyba nawet oddzielny task mial byc)